packages/versioning/src/validate.ts

```diff
@@ -772,6 +773,28 @@ function validateAvailabilityForRef(
   }
 }
 
+function canIgnoreDependentVersioning(type: Type, versioning: "added" | "removed") {
+  if (type.kind === "ModelProperty") {
+    return canIgnoreVersioningOnProperty(type, versioning);
+  }
+  return false;
+}
+
+function canIgnoreVersioningOnProperty(
+  prop: ModelProperty,
+  versioning: "added" | "removed"
+): boolean {
+  if (prop.sourceProperty === undefined) {
+    return false;
+  }
+  // Check if the decorator was defined on this property or a source property. If source property ignore.
+  const selfDecorators = prop.decorators.filter((x) => x.definition?.name === `@${versioning}`);
+  const sourceDecorators = prop.sourceProperty.decorators.filter(
+    (x) => x.definition?.name === `@${versioning}`
```

Comment:
Do we need to take the namespace into account as well in case someone ever writes their own `@added`/`@removed` decorators?

Reply:
yeah that is a good point

Reply:
updated to check against the decorator implementation function